include/deal.II/distributed/shared_tria.h

```diff
@@ -310,6 +331,25 @@ namespace parallel
       void
       load(Archive &ar, const unsigned int version);
 
+      /**
+       * Specify which information will be returned when calling
+       * CellAccessor::subdomain_id() on active cells. The subdomain
+       * modification will only be applied to active cells.
+       *
+       * The subdomain mask should only be changed for temporary read-only
+       * purposes. For example, whenever your implementation requires to treat
+       * artificial cells temporarily as locally relevant to access their dof
+       * indices.
+       *
+       * Modifications to the subdomain ID of active cells by calling
+       * CellAccessor::set_subdomain_id() should always be made when working
+       * with the default mask, i.e., SubdomainMask::default_ids. Otherwise,
+       * changes made to the subdomain IDs could be lost and lead to undefined
+       * behaviour.
+       */
+      void
+      set_subdomain_mask(const SubdomainMask new_mask);
```

Comment:
How can you restore the flag if there is no function that allows you to get the current state?